lib/location/loc_core.c

```diff
@@ -0,0 +1,391 @@
+		return -EINVAL;
+	}
+
+	for (int i = 0; i < config->methods_count; i++) {
+		/* Check if the method is valid */
+		method_api = loc_method_api_validity_get(config->methods[i].method);
+		if (method_api == NULL) {
+			LOG_ERR("Location method (%d) not supported", config->methods[i].method);
+			return -EINVAL;
+		}
+		/* Validate method specific parameters */
+		if (methods_supported[i]->validate_params) {
+			err = methods_supported[i]->validate_params(&config->methods[i]);
+			if (err) {
+				return err;
+			}
+		}
+	}
+	return 0;
+}
+
+static void current_loc_config_set(const struct loc_config *config)
+{
+	assert(config->methods_count <= LOC_MAX_METHODS);
+
+	if (config == &current_loc_config) {
+		/* This is hit in periodic mode */
+		return;
+	}
+
+	current_loc_config.methods_count = config->methods_count;
+	current_loc_config.interval = config->interval;
+
+	memset(current_loc_config.methods, 0,
+		sizeof(struct loc_method_config) * LOC_MAX_METHODS);
+	memcpy(current_loc_config.methods, config->methods,
+		sizeof(struct loc_method_config) * config->methods_count);
+}
+
+int loc_core_location_get(const struct loc_config *config)
+{
+	int err;
+	enum loc_method requested_loc_method;
+
+	struct loc_config default_config = { 0 };
+	struct loc_method_config methods[2] = { 0 };
+
+	if (config == NULL) {
+		LOG_DBG("No configuration given. Using default configuration.");
+
+		default_config.interval = 0; /* Single position */
+
+		default_config.methods_count = 2;
+		default_config.methods = methods;
+
+		default_config.methods[0].method = LOC_METHOD_GNSS;
+		default_config.methods[0].config.gnss.timeout = 120;
+		default_config.methods[0].config.gnss.accuracy = LOC_ACCURACY_NORMAL;
+
+		default_config.methods[1].method = LOC_METHOD_CELLULAR;
+		config = &default_config;
+	}
+
+	current_loc_config_set(config);
+	/* Location request starts from the first method */
+	current_loc_method_index = 0;
+	requested_loc_method = config->methods[current_loc_method_index].method;
+	LOG_DBG("Requesting location with '%s' method",
+		(char *)loc_method_api_get(requested_loc_method)->method_string);
+	loc_core_current_event_data_init(requested_loc_method);
+	err = loc_method_api_get(requested_loc_method)->location_get(
+		&config->methods[current_loc_method_index]);
+
+	return err;
+}
+
+void loc_core_event_cb_error(void)
+{
+	current_event_data.id = LOC_EVT_ERROR;
+
+	loc_core_event_cb(NULL);
+}
+
+void loc_core_event_cb_timeout(void)
+{
+	current_event_data.id = LOC_EVT_TIMEOUT;
+
+	loc_core_event_cb(NULL);
+}
+
+void loc_core_event_cb(const struct loc_location *location)
+{
+	char temp_str[16];
+	enum loc_method requested_loc_method;
+	enum loc_method previous_loc_method;
+	int err;
+
+	if (location != NULL) {
+		/* Location was acquired properly, finish location request */
+		current_event_data.id = LOC_EVT_LOCATION;
+		current_event_data.location = *location;
+
+		LOG_DBG("Location acquired successfully:");
```

Comment:
Should we also log google maps link to the coordinates? Requires 30-50 bytes more from the stack.

Reply:
This has been done